packages/adp-tooling/src/common/endpoints-manager.ts

```diff
@@ -0,0 +1,181 @@
+import { isAppStudio } from '@sap-ux/btp-utils';
+import type { ToolsLogger } from '@sap-ux/logger';
+import type { Endpoint } from '@sap-ux/environment-check';
+import { getCredentialsFromStore } from '@sap-ux/system-access';
+import { checkEndpoints, isExtensionInstalledVsCode } from '@sap-ux/environment-check';
+
+import type { SystemDetails } from '../types';
+
+/**
+ * Determines if authentication is not required for accessing a specified system.
+ *
+ * @param {string} system - The name of the system to check.
+ * @param {Endpoint[]} endpoints Array of system endpoints.
+ * @returns {boolean} True if no authentication is required, false otherwise.
+ */
+function getDestinationRequiresAuth(system: string, endpoints: Endpoint[]): boolean {
+    const found = endpoints.find((endpoint: Endpoint) => endpoint.Name === system);
+
+    return found?.Authentication === 'NoAuthentication';
+}
+
+/**
+ * Checks if a specified system requires authentication based on the endpoint information and installation status.
+ *
+ * @param {string} system - The name of the system to check.
+ * @param {Endpoint[]} endpoints Array of system endpoints.
+ * @returns {boolean} True if the system requires authentication, false otherwise.
+ */
+function getSystemRequiresAuthentication(system: string, endpoints: Endpoint[]): boolean {
+    const isInstalled = isExtensionInstalledVsCode('sapse.sap-ux-application-modeler-extension');
+    if (!isInstalled || endpoints.length === 0) {
+        return true;
+    }
+
+    return !(
+        endpoints.filter((endpoint) => endpoint.Url === system).length > 0 ||
+        endpoints.filter((endpoint) => endpoint.Name === system).length > 0
+    );
+}
+
+/**
+ * Retrieves the names of all stored endpoints and sorted alphabetically.
+ *
+ * @param {Endpoint[]} endpoints Array of system endpoints.
+ * @returns {string[]} An array of endpoint names and sorted order.
+ */
+export function getEndpointNames(endpoints: Endpoint[]): string[] {
+    return endpoints
+        .map((endpoint) => endpoint.Name)
+        .sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase(), 'en', { sensitivity: 'base' }));
+}
+
+/**
+ * Service class to manage and retrieve information about system endpoints,
+ * including their names, authentication requirements, and specific details.
+ */
+export class EndpointsManager {
+    private static instance: EndpointsManager;
+    private endpoints: Endpoint[];
+    private isExtensionInstalled: boolean;
+
+    /**
+     * Creates an instance of EndpointsManager.
+     *
+     * @param {ToolsLogger} logger - The logger.
+     */
+    private constructor(private logger: ToolsLogger) {
+        this.endpoints = [];
+        this.isExtensionInstalled = isExtensionInstalledVsCode('sapse.sap-ux-application-modeler-extension');
```

Comment:
the extension sapse.sap-ux-application-modeler-extension is not a pre-requisite here. The extension allows modifying the stored systems, however, the systems itself are stored in the OS secure storage and can be added/manipulated with other tools as well, therefore, please remove that check.

Reply:
Then we need to check whether OS secure storage exists? Otherwise, if it doesn't, then how can we understand what prompts to show. If it exists then we need a list prompt with all the systems. If it doesn't exist then we show input prompts for system, client, username and password.

Reply:
Just call the store module and ask it for a list of systems. If at least one is returned you are good to go would be an option. The better option is that to reuse the functionality that is currently being implemented in the `odata-service-inquirer` because that one always offers a `new system` choice that then triggers the system creation prompts.

Reply:
Fixed in [ad544c4](https://github.com/SAP/open-ux-tools/pull/2169/commits/ad544c432e18945f92cb07099b155a4474eec626)